client/ayon_nuke/api/lib.py

```diff
@@ -810,89 +812,24 @@ def check_inventory_versions():
     and check if the node is having actual version. If not then it will color
     it to red.
     """
-    from .pipeline import parse_container
-
-    # get all Loader nodes by avalon attribute metadata
-    node_with_repre_id = []
-    repre_ids = set()
-    # Find all containers and collect it's node and representation ids
-    for node in nuke.allNodes():
-        container = parse_container(node)
-
-        if container:
-            node = nuke.toNode(container["objectName"])
-            avalon_knob_data = read_avalon_data(node)
-            repre_id = avalon_knob_data["representation"]
-
-            repre_ids.add(repre_id)
-            node_with_repre_id.append((node, repre_id))
-
-    # Skip if nothing was found
-    if not repre_ids:
-        return
-
+    host = registered_host()
+    containers = host.get_containers()
     project_name = get_current_project_name()
-    # Find representations based on found containers
-    repre_entities = ayon_api.get_representations(
-        project_name,
-        representation_ids=repre_ids,
-        fields={"id", "versionId"}
-    )
-    # Store representations by id and collect version ids
-    repre_entities_by_id = {}
-    version_ids = set()
-    for repre_entity in repre_entities:
-        # Use stringed representation id to match value in containers
-        repre_id = repre_entity["id"]
-        repre_entities_by_id[repre_id] = repre_entity
-        version_ids.add(repre_entity["versionId"])
-
-    version_entities = ayon_api.get_versions(
-        project_name,
-        version_ids=version_ids,
-        fields={"id", "version", "productId"},
-    )
-    # Store versions by id and collect product ids
-    version_entities_by_id = {}
-    product_ids = set()
-    for version_entity in version_entities:
-        version_entities_by_id[version_entity["id"]] = version_entity
-        product_ids.add(version_entity["productId"])
-
-    # Query last versions based on product ids
-    last_versions_by_product_id = ayon_api.get_last_versions(
-        project_name, product_ids=product_ids, fields={"id", "productId"}
-    )
-
-    # Loop through collected container nodes and their representation ids
-    for item in node_with_repre_id:
-        # Some python versions of nuke can't unfold tuple in for loop
-        node, repre_id = item
-        repre_entity = repre_entities_by_id.get(repre_id)
-        # Failsafe for not finding the representation.
-        if not repre_entity:
-            log.warning((
-                "Could not find the representation on node \"{}\""
-            ).format(node.name()))
-            continue
 
-        version_id = repre_entity["versionId"]
-        version_entity = version_entities_by_id.get(version_id)
-        if not version_entity:
-            log.warning((
-                "Could not find the version on node \"{}\""
-            ).format(node.name()))
-            continue
+    category_colors = {
```

Comment:
do we have any better place where we could define the different generic colors to use for these different categories?

I made `invalid` red and `not_found` as yellow... but previously we didn't do anything to color those, do we want to keep it that way?

Reply:
You could move these to project settings if they feel like something that should be customizable for a studio. Otherwise as it is now or as contants should be fine. I'd keep it as is until a feature request to customize it comes in - that way, the implementation for now remains light-weight.

Reply:
yeah I agree that is out of scope and should be a separate PR. Having that said, do you like the color palette chosen for the `not_found` and `invalid`?

Reply:
> yeah I agree that is out of scope and should be a separate PR. Having that said, do you like the color palette chosen for the not_found and invalid?

Not a Nuke user myself ;) so can't test it visually inside of Nuke. I'll leave this to @jakubjezek001

Reply:
I really like the approach! Thanks @fabiaserra. I wonder if we could just add the `category_colors` into **api/constants.py** for now.

Reply:
Yeah, this is the best we are having now.